src/tool_formparse.c

```diff
@@ -501,7 +501,7 @@ static int get_param_part(struct OperationConfig *config, char endchar,
       type = p;
 
       /* find end of content-type */
-      while(*p && (ISALPHA(*p) || (*p == '/') || (*p == '-')))
+      while(*p && (ISALNUM(*p) || (*p == '/') || (*p == '-')))
```

Comment:
From what I read in the RFC if content type or subtype is an extension-token ("X-" prefixed type) that can be any `token`. I think such a cursory check doesn't need strict RFC adherence however the old sscanf behavior accepted more and that was more in line with the RFC. 

```suggestion
      while(*p && *p > 32 && *p < 127 && !strchr("()<>@,;:\\\"/[]?=", *p))
```
Note in the suggestion above, the scan for banned characters could easily be put in the loop instead of calling strchr for each character if you prefer like `for(banned = "()<>@,;:\\\"/[]?="; *banned; ++banned) {`.

~~~
                                                 ; (  Octal, Decimal.)
     CHAR        =  <any ASCII character>        ; (  0-177,  0.-127.)
     CTL         =  <any ASCII control           ; (  0- 37,  0.- 31.)
                     character and DEL>          ; (    177,     127.)

     token := 1*<any (US-ASCII) CHAR except SPACE, CTLs,
                 or tspecials>
     tspecials :=  "(" / ")" / "<" / ">" / "@" /
                   "," / ";" / ":" / "\" / <">
                   "/" / "[" / "]" / "?" / "="
                   ; Must be in quoted-string,
                   ; to use within parameter values
~~~

Ref: https://datatracker.ietf.org/doc/html/rfc822#section-3.3
Ref: https://datatracker.ietf.org/doc/html/rfc2045#section-5.1

Reply:
I think strcspn() is good for this.